packages/cactus-plugin-keychain-vault/src/main/typescript/plugin-keychain-vault.ts

```diff
@@ -97,15 +109,30 @@ export class PluginKeychainVault implements IPluginWebService, IPluginKeychain {
     this.endpoint = this.opts.endpoint;
     this.apiVersion = this.opts.apiVersion || "v1";
 
-    this.kvSecretsMountPath =
-      opts.kvSecretsMountPath || K_DEFAULT_KV_SECRETS_MOUNT_PATH;
-    this.log.info(`this.kvSecretsMountPath=${this.kvSecretsMountPath}`);
+    const kvSupport = opts.kvSecretsMountPath !== undefined;
+    const transitSupport = opts.transitSecretMountPath !== undefined;
+    if (!kvSupport && !transitSupport) {
+      throw new Error("require at least kv mount path or transit mount path");
+    }
+    if (kvSupport) {
```

Comment:
@Zzocker Is it possible that these three if conditions could be refactored into a single `if {  } if else {} else { throw }` where by the end of it the `backend` field is either definitely assigned (e.g. no need for the `?` optional marker) OR we have thrown an exception if no transit and no kv support was provided.
I'm just asking this because I'm not sure if this would cause complications in the rest of the implementation or not. If it would, then let's discuss further. If it wouldn't then I think it would be a great simplification and also we could guarantee this way that if the vault plugin instance was created, then it definitely has a valid `backend` so no need to check for it in all the methods that use the `backend` variable.

Reply:
sounds reasonable , I am using two different backend (one for each `kv` and `transit`) because , I wanted to take advantage of `node-vault` package functionality of specifying montPath in `pathPrefix` filed while creating the `backend`. so that I won't have to implement my own function for creating `path` just like you did [here](https://github.com/hyperledger/cactus/blob/4cc6f2c5d3345a7af8cc04b9218c38d7670872a8/packages/cactus-plugin-keychain-vault/src/main/typescript/plugin-keychain-vault.ts#L219) .

But I think if adding one liner function for creating transitPath can simplify `if-else` in constructor, than I should add that function and  simplify the constructor